processor/src/main/java/org/mapstruct/ap/internal/util/accessor/Accessor.java

```diff
@@ -26,17 +26,39 @@
 import javax.lang.model.type.TypeMirror;
 
 /**
+ * This represents an Accessor that can be used for writing/reading a property to/from a bean.
+ *
  * @author Filip Hrisafov
  */
 public interface Accessor {
 
+    /**
+     * This returns the type that this accessor gives as a return.
+     *
+     * e.g. The {@link ExecutableElement#getReturnType()} if this is a method accessor,
+     * or {@link javax.lang.model.element.VariableElement#asType()} for field accessors.
+     *
+     * @return the type that the accessor gives as a return
+     */
```

Comment:
If the accessor is for writing a property, this could also be the argument type of a setter method?